File: app/src/main/java/com/lin/linreader/http/ZhiHuApi.java
package com.lin.linreader.http;

import com.lin.linreader.bean.Stories;
import com.lin.linreader.bean.ZhiHuDailyBean;

import java.util.List;

import rx.Observable;
import rx.Subscriber;

/**
 * Created by lin on 2018/2/2.
 */

public class ZhiHuApi {
    public static ZhiHuApi mZhiHuApi;
    public ZhiHuService userService;
    private BaseApi mBaseApi;

    public ZhiHuApi(int code) {
        mBaseApi = new BaseApi();
        userService = MyHttpMethod.Companion.getInstance(1).createApi(ZhiHuService.class);
    }

    public static ZhiHuApi getInstance(int code) {
        switch (code) {
            case 1:
                if (mZhiHuApi == null) {
                    synchronized (ZhiHuApi.class) {
                        if (null == mZhiHuApi)
                            mZhiHuApi = new ZhiHuApi(code);
                    }
                }
                return mZhiHuApi;
        }
        return null;
    }
    /*-------------------------------------获取的方法-------------------------------------*/

    public void getZhiHuListInfo(String date, Subscriber<List<Stories>> subscriber) {
        Observable observable = userService.getZhiHuList(date)
                .map(new BaseApi.HttpResultFunc<ZhiHuDailyBean>());

        mBaseApi.toSubscribe(observable, subscriber);
    }


}

File: app/src/main/java/com/lin/linreader/http/ZhiHuApi.kt
package com.lin.linreader.http

import com.lin.linreader.bean.ZhiHuDailyBean
import rx.Subscriber

/**
 * Created by lin on 2018/2/2.
 */

class ZhiHuApi(code: Int) : BaseApi() {
    var userService: ZhiHuService

    init {
        userService = MyHttpMethod.getInstance(code)!!.createApi(ZhiHuService::class.java)
    }
    /*-------------------------------------获取的方法-------------------------------------*/

    fun getZhiHuListInfo(date: String, subscriber: Subscriber<ZhiHuDailyBean>) {
        val observable = userService.getZhiHuList(date)
                .map(BaseApi.HttpResultFunc())

        toSubscribe(observable, subscriber)
    }

    companion object {
        var mZhiHuApi: ZhiHuApi? = null

        fun getInstance(code: Int): ZhiHuApi? {
            when (code) {
                1 -> {
                    if (mZhiHuApi == null) {
                        synchronized(ZhiHuApi::class.java) {
                            if (null == mZhiHuApi)
                                mZhiHuApi = ZhiHuApi(code)
                        }
                    }
                    return mZhiHuApi
                }
            }
            return null
        }
    }


}

